packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt

```diff
@@ -0,0 +1,58 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package expo.modules.kotlin.types
+
+import kotlin.reflect.KClass
+
+open class Either<FirstType : Any, SecondType : Any>(
+  @PublishedApi
+  internal val value: Any
+) {
+  @JvmName("isFirstType")
+  fun `is`(type: KClass<FirstType>): Boolean {
+    return type.isInstance(value)
+  }
+
+  @JvmName("isSecondType")
+  fun `is`(type: KClass<SecondType>): Boolean {
+    return type.isInstance(value)
+  }
+
+  @JvmName("getFirstType")
+  fun get(type: KClass<FirstType>) = value as FirstType
+
+  @JvmName("getSecondType")
+  fun get(type: KClass<SecondType>) = value as SecondType
+
+  fun first() = value as FirstType
```

Comment:
Not a bad idea to add such functions. One thing that I would be worried about is that if we're to merge those four types together into one generic `Either`, then we will probably have to deprecate this. It's nothing to worry about right now, so let's land it as is.